firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java

```diff
@@ -458,6 +458,28 @@ public ListenerRegistration addSnapshotListener(
         Executors.DEFAULT_CALLBACK_EXECUTOR, internalOptions(metadataChanges), activity, listener);
   }
 
+  /**
+   * Starts listening to the document referenced by this {@code DocumentReference} with the given
+   * options.
+   *
+   * @param options Sets snapshot listen options, including whether metadata-only changes should
+   *     trigger snapshot events, the source that listens to, the executor to use to call the
```

Comment:
Minor edits:
+ (listen -> listener) "...Sets snapshot listener options...."
+ "...the source to listen to...."